core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -874,6 +874,11 @@ public int performBatchRecovery(Iterable<Translog.Operation> operations) {
      * After the store has been recovered, we need to start the engine in order to apply operations
      */
     public void performTranslogRecovery(boolean indexExists) {
+        if (indexExists == false) {
+            // note: these are set when recovering from the translog
+            recoveryState().getTranslog().totalOperations(0);
```

Comment:
any chance we can assign the `recoveryState().getTranslog()` somewhere and don't chain the calls all the time?